crates/ndc-clickhouse/src/sql/query_builder.rs

```diff
@@ -198,74 +198,81 @@ impl<'r, 'c> QueryBuilder<'r, 'c> {
-                                ]);
-                                Function::new_unquoted("COUNT")
-                                    .args(vec![column.into_arg()])
-                                    .distinct(*distinct)
-                                    .into_expr()
-                            }
-                            models::Aggregate::SingleColumn {
-                                function,
-                                column: _,
-                                field_path: _,
-                            } => {
-                                let column = Expr::CompoundIdentifier(vec![
-                                    Ident::new_quoted("_row"),
-                                    Ident::new_quoted(format!("_agg_{alias}")),
-                                ]);
-                                apply_function(&aggregate_function(function)?, column)
+            Some(
+                if aggregates.is_empty() {
+                    Function::new_unquoted("map").into_expr()
+                } else {
+                    let args = aggregates
+                        .iter()
+                        .map(|(alias, aggregate)| {
+                            Ok(match aggregate {
+                                models::Aggregate::StarCount {} => Function::new_unquoted("COUNT")
+                                    .args(vec![FunctionArgExpr::Wildcard.into_arg()])
+                                    .into_expr(),
+                                models::Aggregate::ColumnCount {
+                                    distinct,
+                                    column: _,
+                                    field_path: _,
+                                } => {
+                                    let column = Expr::CompoundIdentifier(vec![
+                                        Ident::new_quoted("_row"),
+                                        Ident::new_quoted(format!("_agg_{alias}")),
+                                    ]);
+                                    Function::new_unquoted("COUNT")
+                                        .args(vec![column.into_arg()])
+                                        .distinct(*distinct)
+                                        .into_expr()
+                                }
+                                models::Aggregate::SingleColumn {
+                                    function,
+                                    column: _,
+                                    field_path: _,
+                                } => {
+                                    let column = Expr::CompoundIdentifier(vec![
+                                        Ident::new_quoted("_row"),
+                                        Ident::new_quoted(format!("_agg_{alias}")),
+                                    ]);
+                                    apply_function(&aggregate_function(function)?, column)
+                                }
```

Comment:
This above is just a format change right?

Reply:
mostly formatting yes. there's an actual change at the bottom, and I guess cargo decided that warranted a reformat